tests/ut/ir/statements/test_scope_stmt.py

```
@@ -80,9 +77,31 @@ def test_scope_stmt_default_name_is_empty(self):
-        scope = ir.ScopeStmt(ir.ScopeKind.InCore, body, span)
+        scope = ir.InCoreScopeStmt(body=body, span=span)
         assert scope.name_hint == ""
 
+    def test_spmd_scope_requires_positive_core_num(self):
+        """SpmdScopeStmt enforces core_num > 0 at construction (compile-time guarantee for type)."""
+        span = ir.Span("test.py", 1, 1, 1, 10)
+        var_x = ir.Var("x", ir.TensorType([64], DataType.FP32), span)
+        var_y = ir.Var("y", ir.TensorType([64], DataType.FP32), span)
+        body = ir.AssignStmt(var_y, var_x, span)
+
+        with pytest.raises(Exception):
```

Ruff (0.15.10)

[warning] Do not assert blind exception: `Exception`

(B017)